runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -603,13 +606,40 @@ InterpreterEmulator::visitInvokedynamic()
    TR::Node *callNode = 0;
    TR::ResolvedMethodSymbol *resolvedSymbol = 0;
    Operand *result = NULL;
-
-   TR_ResolvedMethod * owningMethod = _methodSymbol->getResolvedMethod();
+   int32_t callSiteIndex = next2Bytes();
    TR::KnownObjectTable *knot = comp()->getOrCreateKnownObjectTable();
-   if (knot && !owningMethod->isUnresolvedCallSiteTableEntry(cpIndex))
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   if (!knot && owningMethod->isUnresolvedCallSiteTableEntry(callSiteIndex)) return;
+
+   // add appendix object to knot and push to stack
+   uintptr_t appendixAddress = owningMethod->appendixAddressFromInvokeDynamicSideTable(callSiteIndex);
+   push(new (trStackMemory()) KnownObjOperand(knot->getOrCreateIndexAt(&appendixAddress)));
+
+   TR_J9VMBase *fej9 = comp()->fej9();
+   TR_OpaqueMethodBlock* targetMethodObj = 0;
+      {
+      TR::VMAccessCriticalSection vmAccess(fej9);
+      targetMethodObj = fej9->targetMethodFromMemberName(owningMethod->memberNameAddressFromInvokeDynamicSideTable(callSiteIndex));
```

Comment:
If `memberNameAddressFromInvokeDynamicSideTable` returns an address, we need to dereference the address to get the MemberName. Or you can add a function `targetMethodFromMemberNameAt` to take address of MemberName, and return you the target method. Inside `targetMethodFromMemberNameAt`, you need to acquire vm access